src/main/java/io/api/etherscan/core/impl/BasicProvider.java

```diff
@@ -42,7 +46,20 @@
         try {
             return gson.fromJson(json, tClass);
         } catch (Exception e) {
-            throw new ParseException(e.getMessage(), e.getCause());
+            if (e instanceof JsonSyntaxException) {
+                Map<String, Object> map = gson.fromJson(json, Map.class);
+                Object statusCode = map.get("status");
```

Comment:
What is the point to check for status & message if you only consider result field in the end?